crates/sock-prot/src/protector/mod.rs

```diff
@@ -0,0 +1,44 @@
+#[cfg(windows)]
+#[path = "windows.rs"]
+mod platform;
+
+#[cfg(any(target_os = "macos", target_os = "ios", target_os = "tvos"))]
+#[path = "apple.rs"]
+mod platform;
+
+#[cfg(target_os = "linux")]
+#[path = "linux.rs"]
+mod platform;
+
+#[cfg(target_os = "android")]
+#[path = "unsupported.rs"]
+mod platform;
+
+use std::io;
+
+pub use platform::NativeProtector;
+
+use crate::NativeSocket;
+
+pub trait Protector: Send + Sync {
+    /// # Safety
+    /// The socket descriptor must be valid
+    unsafe fn make_external(&self, _: NativeSocket) -> io::Result<()> {
```

Comment:
why has this become unsafe?

Reply:
It takes in raw file descriptors, the caller needs to guarantee that this is a proper thing

Reply:
I'm not sure I understand. If someone passes calls `make_external` with a instance of `NativeSocket` - are there any additional requirements that need to be true for the `make_external` to work correctly that the compiler is not verifying?

Reply:
Yeah, since `NativeSocket` is `i32` (of Linux at least) then the caller needs to ensure this `i32` is a valid socket FD and not for example a file on disk or stdin (0)